lib/src/components/NxSubmitMask/NxSubmitMask.tsx

```diff
@@ -28,9 +30,15 @@ const NxSubmitMask: FunctionComponent<Props> =
 
     return (
       <div className={classes}>
-        <h3 className="nx-h3 nx-submit-mask__message">
-          <NxLoadingSpinner>{spinnerMessage}</NxLoadingSpinner>
-        </h3>
+        <div className="nx-submit-mask__message">
+          { success ?
+            <>
+              <NxFontAwesomeIcon icon={faCheckCircle} className="nx-loading-spinner__icon" />
+              <span>{spinnerMessage || 'Loading…'}</span>
+            </> :
+            <NxLoadingSpinner>{spinnerMessage}</NxLoadingSpinner>
```

Comment:
I think the variables are a little bit tangled up.  What you have works but it uses variables names that aren't quite accurate to what the variables do.  What I'd do instead is pass the `message` prop directly to the `<NxLoadingSpinner>`, and then instead of having a variable called `spinnerMessage`, use the `successMessage` prop in the JSX construction for the success case, with the `|| 'Success!'` part in there for the default.

Reply:
> I think the variables are a little bit tangled up. What you have works but it uses variables names that aren't quite accurate to what the variables do. What I'd do instead is pass the `message` prop directly to the `<NxLoadingSpinner>`, and then instead of having a variable called `spinnerMessage`, use the `successMessage` prop in the JSX construction for the success case, with the `|| 'Success!'` part in there for the default.

I did a bunch of work on it here, I think it's much better now: https://github.com/sonatype/sonatype-react-shared-components/pull/182/commits/7eaad8766f8001a37f8fff6f890d10ea7a8be8af